include/dd/FunctionalityConstruction.hpp

```diff
@@ -9,11 +9,15 @@ using namespace qc;
 
 template <class Config>
 MatrixDD buildFunctionality(const QuantumComputation* qc,
-                            std::unique_ptr<dd::Package<Config>>& dd);
+                            std::unique_ptr<dd::Package<Config>>& dd,
+                            bool reduceAncillaeQubits = true,
```

Comment:
I called it "ancillae" in order to be consistent with the name of the "reduceAncillae" function. Maybe it would make more sense to instead rename the "zeroAncillaPartialEquivalence" to "zeroAncillaePartialEquivalence"?

Reply:
"Ancillae" is just the (latin) plural of "Ancilla". 
One can either refer to "Ancillae" or to "Ancilla Qubits" (or "Ancillary Qubits").
I believe at the moment we use all different versions.
Bringing some consistency into this would be appreciated.
I still think it should be either "reduceAncillaQubits" or "reduceAncillae" for the parameter above, but I agree with your proposal of "zeroAncillaePartialEquivalence".

Reply:
Ah ok I understand, "ancilla" or "ancillary" are adjectives and "ancillae" is the plural noun. Then I think it would be even more consistent with the already existing functions if I call it "reduceAncillaryQubits" (I can't call it "reduceAncillae", because there is already a function with this name).

Reply:
That suggestion sounds good to me 👍🏼